support/builder/build_dep_populate.sh

```diff
@@ -0,0 +1,463 @@
+    echo "###########################################################"
+    echo "==========================================================="
+    echo ""
+    echo "This tool will scrape all the hart files from an s3 or minio"
+    echo "bucket and reingest them to populate build dep and tdep metadata"
+    credsConfigured=false
+
+    setBldrUrl
+
+
+
+    if [[ -n "${HAB_AUTH_TOKEN+x}" ]]; then
+        echo ""
+        echo "We were able to discover your builder auth token as: "
+        echo "${HAB_AUTH_TOKEN}"
+        read -r -p "Would you like to use these credentials? [y/N]" response
+            if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                echo "Setting detected credentials"
+            fi
+    else
+        echo ""
+        echo "WARNING: No Builder credentials detected!"
+        read -r -p "Would you like to generate them now? [y/N] " response
+        if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+            genBldrCreds
+        else
+            echo "Please configure your builder auth-token and re-run s3migrate."
+            echo ""
+            exit 1
+        fi
+    fi
+
+    if [ "$s3type" = "minio" ]; then
+        echo ""
+        echo "It looks like you specified an ingestion from minio!"
+
+        sgroups=("default" "dev" "prod")
+        for i in "${sgroups[@]}"; do
+            if curl -s localhost:9631/services/builder-minio/"${i}" | jq .cfg.key_id > /dev/null; then
+                access_key_id=$(curl -s localhost:9631/services/builder-minio/default | jq .cfg.key_id)
+                secret_access_key=$(curl -s localhost:9631/services/builder-minio/default | jq .cfg.secret_key)
+                echo ""
+                echo "We were able to detect your minio credentials!"
+                echo "(ACCESS_KEY_ID) Username: ${access_key_id}"
+                echo "(SECRET ACCESS_KEY) Password: ${secret_access_key}"
+                echo ""
+                read -r -p "Would you like to use these credentials? [y/N] " response
+                    if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                        echo "Setting detected credentials"
+                        export AWS_ACCESS_KEY_ID=${access_key_id//\"}
+                        export AWS_SECRET_ACCESS_KEY=${secret_access_key//\"}
+                        return
+                    else
+                        credCheck
+                    fi
+            else
+                echo ""
+                echo "Minio will use whatever credentials you've configured it with."
+                echo "If those credentials don't match your aws credentials file, you"
+                echo "must specify those custom credentials."
+                credCheck
+            fi
+        done
+    else
+        credCheck
+    fi
+
+}
+
+artifacts=()
+
+if [[ -z ${1:-} ]]; then
+    echo "Invalid Argument. Argument must be either 'minio' or 'aws'"
+    exit 1
+fi
+
+case ${1} in
```

Comment:
Change this to
```suggestion
case "${1:-}" in
```
and the above check becomes redundant